File: LogisimPlus/LogisimPlus/draw/toolbar/ToolbarItem.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

using LogisimPlus.Java;

namespace draw.toolbar
{

	public interface ToolbarItem
	{
		bool Selectable {get;}

		void paintIcon(JComponent destination, JGraphics g);

		string ToolTip {get;}

		Size getSize(object orientation);
	}

}

File: logisim/app/src/main/java/draw/toolbar/ToolbarItem.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package draw.toolbar;

import java.awt.Component;
import java.awt.Dimension;
import java.awt.Graphics;

public interface ToolbarItem {
	boolean isSelectable();

	void paintIcon(Component destination, Graphics g);

	String getToolTip();

	Dimension getDimension(Object orientation);
}
